src/main/drivers/dshot.c

```diff
@@ -164,3 +164,48 @@ void updateDshotTelemetryQuality(dshotTelemetryQuality_t *qualityStats, bool pac
 #endif // USE_DSHOT_TELEMETRY_STATS
 
 #endif // USE_DSHOT
+
+// temporarly here, needs to be moved during refactoring
+
+void fixReorderingArray(uint8_t *array, const unsigned size)
```

Comment:
Not quite sure what this does. But the approach taken in the configuration validation is that if an invalid value is found, the configuration is reset to (safe) defaults, and this is what this should be doing.

Reply:
This one is for validateAndFixConfiguration() as you suggested to fix any possible invalid order.
It fixes  weird order like 1,2,20,3 to 1,2,4,3
Or 1,2,2,3 to 1,2,4,3, or even 100, 100, 3, 4 to 1,2,3,4.

The default values are 1,2,3,4,... and that's in pgResetFn_motorConfig()

Reply:
> The default values are 1,2,3,4,... and that's in pgResetFn_motorConfig():

Yes, so this should reset the values to the default (1, 2, 3, 4,...), and not some arbitrary ordering that is not what the user entered.

Reply:
@mikeller  so you mean that in ANY weird case like **4,3,2,1,7,7,7,7** it should just reset to 0,1,2,3,4,5,6,7?

Reply:
@mikeller  I renamed that function to `validateAndfixMotorOutputReordering()`, now it is resetting the array to 0,1,2,3... if there is something wrong detected with it.
Also changed unit tests.
Thanks!